src/java/org/apache/cassandra/exceptions/RepairException.java

```diff
@@ -27,9 +29,9 @@ public class RepairException extends Exception
 {
     private final boolean shouldLogWarn;
 
-    private RepairException(RepairJobDesc desc, PreviewKind previewKind, String message, boolean shouldLogWarn)
+    private RepairException(@Nullable RepairJobDesc desc, PreviewKind previewKind, String message, boolean shouldLogWarn)
     {
-        this(desc.toString(previewKind != null ? previewKind : PreviewKind.NONE) + ' ' + message, shouldLogWarn);
+        this((desc == null ? "" : desc.toString(previewKind != null ? previewKind : PreviewKind.NONE)) + ' ' + message, shouldLogWarn);
```

Comment:
nit: Have to actually read more to figure out where the nulls can come from, but if we control that, I wonder if there's an alternative. Always a little worry of allowing `null` in general when `RepairJobDesc` is usually not supposed to be...

Reply:
I forget, but there was 1 specific code path that NPE due to this... so I just documented... ill try to find the call path again

Reply:
Ok, found the issue.  Not all `RepairMessage` have a `desc`

```
public class CleanupMessage extends RepairMessage
{
    public final TimeUUID parentRepairSession;

    public CleanupMessage(TimeUUID parentRepairSession)
    {
        super(null);
        this.parentRepairSession = parentRepairSession;
    }
```

this leads to NPE

Reply:
> I wonder if there's an alternative

some of the code that calls this path only know `RepairMessage`, so seems tricky as we need type checks or something to solve this without using null?